### DISCUSSION TITLE
Styling - highlight each char individually

### DISCUSSION THREAD

**Discussion by jacek-kurlit** [ratatui/ratatui#1451] [Q&A] [answered]
Hi,

I'm working on tool called [pik](https://github.com/jacek-kurlit/pik)

It allows for fuzzy search of running processes and I wanted to add matched text highlights.
I know that ratatui can style text but my problem is that I want to change style of individual characters not entire words and now I'm looking for best approach to do it.

Let's say I have text "/usr/lib/firefox/firefox" and I want to highlight only text in brackets ->  "/usr/lib/(f)ire(fox)/firefox"
Should I then create a Line build from multiple spans?
Like so (simplified example, not real code):
```
line = vec![Span("/usr/lib/", White), Span("f", Yellow), Span("ire", White), Span("fox", Yellow), Span("/firefox", White)]
```
Is this best approach? Is there any other option?

Thanks!

**Comment by joshka** [comment, marked as answer]
>Is this best approach? Is there any other option?

Yes. You could also use some sort of replacement on the string with Ansi styles and the use ansi-to-tui to convert to the styled spans, but the level of effort is likely similar.

**Reply by jacek-kurlit** [reply]
Thank you, I will implement it this way.

**Comment by orhun** [comment]
`pik` is super awesome btw, I use it every day!

I implemented a similar search highlighting feature before if it helps: https://github.com/orhun/binsider/blob/19bea73f66eaaeaa0fb5834b25aedc71a280b83f/src/tui/ui.rs#L1142-L1156

**Reply by jacek-kurlit** [reply]
Thanks, I will take a look at this
